src/kernel/emulation/linux/misc/sysctl_machdep.c

```diff
@@ -51,3 +77,132 @@ sysctl_handler(handle_brand_string)
 	return 0;
 }
 
+sysctl_handler(handle_max_basic)
+{
+	setup(0);
+
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+
+	sprintf(old,"%d",eax);
+
+	return 0;
+}
+
+sysctl_handler(handle_family)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 7;
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_model)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax = eax >> 3;
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_stepping)
+{
+    setup(1);
+
+    __cpuid(level, eax, ebx, ecx, edx);
+
+    eax &= 15;
+
+    sprintf(old,"%d",eax);
+
+    return 0;
+}
+
+sysctl_handler(handle_brand_string)
+{
+    unsigned int level = 0;
+    unsigned int eax = 0x80000000;
+
+    unsigned int test[3];
+
+    __cpuid(level,eax,test, test+1,test+2);
```

Comment:
This function has a bunch of "invalid lvalue in asm output" errors when compiling, and this is because you can't use arrays as assembly arguments. What you can do is define `ebx`, `ecx`, and `edx` (like in the other functions) and use those as your arguments.
```suggestion
    unsigned int ebx = 0;
    unsigned int ecx = 0;
    unsigned int edx = 0;
    __cpuid(level, eax, ebx, ecx, edx);
```

Reply:
done (then I went with for loops)